x/crosschain/keeper/keeper_cross_chain_tx_vote_inbound_tx.go

```diff
@@ -138,27 +139,45 @@ func (k msgServer) VoteOnObservedInboundTx(goCtx context.Context, msg *types.Msg
 				cctx.CctxStatus.ChangeStatus(types.CctxStatus_Aborted, "invalid sender chain")
 				return &types.MsgVoteOnObservedInboundTxResponse{}, nil
 			}
-			medianGasPrice, isFound := k.GetMedianGasPriceInUint(ctx, chain.ChainId)
-			if !isFound {
-				cctx.CctxStatus.ChangeStatus(types.CctxStatus_Aborted, "cannot find gas price")
-				return &types.MsgVoteOnObservedInboundTxResponse{}, nil
-			}
 			// create new OutboundTxParams for the revert
 			cctx.OutboundTxParams = append(cctx.OutboundTxParams, &types.OutboundTxParams{
-				Receiver:           cctx.InboundTxParams.Sender,
-				ReceiverChainId:    cctx.InboundTxParams.SenderChainId,
-				Amount:             cctx.InboundTxParams.Amount,
-				CoinType:           cctx.InboundTxParams.CoinType,
-				OutboundTxGasLimit: 0, // for fungible refund, use default gasLimit
-				OutboundTxGasPrice: medianGasPrice.MulUint64(2).String(),
+				Receiver:        cctx.InboundTxParams.Sender,
+				ReceiverChainId: cctx.InboundTxParams.SenderChainId,
+				Amount:          cctx.InboundTxParams.Amount,
+				CoinType:        cctx.InboundTxParams.CoinType,
+				// use same gas limit as outbound
+				//TODO: determine a specific revert gas limit https://github.com/zeta-chain/node/issues/1065
+				OutboundTxGasLimit: msg.GasLimit,
 			})
-			if err = k.UpdateNonce(ctx, chain.ChainId, &cctx); err != nil {
+
+			// we create a new cached context, and we don't commit the previous one with EVM deposit
+			tmpCtx, commit := ctx.CacheContext()
+			err = func() error {
+				err := k.PayGasAndUpdateCctx(
+					tmpCtx,
+					chain.ChainId,
+					&cctx,
+					cctx.InboundTxParams.Amount,
+					false,
+				)
+				if err != nil {
+					return err
+				}
+				err = k.UpdateNonce(tmpCtx, chain.ChainId, &cctx)
+				if err != nil {
+					return err
+				}
+				return nil
+			}()
+			if err != nil {
+				// do not commit anything here as the CCTX should be aborted
 				cctx.CctxStatus.ChangeStatus(types.CctxStatus_Aborted, err.Error())
 				return &types.MsgVoteOnObservedInboundTxResponse{}, nil
 			}
-			// do not commit() here;
+			commit()
 			cctx.CctxStatus.ChangeStatus(types.CctxStatus_PendingRevert, revertMessage)
```

Comment:
Seems we have a few places that `ChangeStatus` of a cctx just in memory only. Is that by intention/design for some reason or these status need to be written to chain state?

Reply:
Haven't touch this logic, if this state change is not committed it might be an issue

Reply:
Yeah, I see. Maybe it's better to confirm with Tanmay/Panruo if it's by design or something.

Reply:
After checking, we actually update it, the code updating the cctx is called in a defer:
https://github.com/zeta-chain/node/blob/037dc0be504e5187adfd96cd5d2fcc098cea9f3c/x/crosschain/keeper/keeper_cross_chain_tx_vote_inbound_tx.go#L124

I would say it's a bit unclear though